core/src/main/java/com/linecorp/armeria/internal/server/annotation/AnnotatedServiceFactory.java

```diff
@@ -503,6 +503,9 @@ private static int order(Method method) {
                             }
                             if (usePathPatterns) {
                                 httpMethodPaths.addAll(pathPatterns);
+                            } else if (httpMethodPaths.isEmpty()) {
+                                // Add an empty value if HTTP method annotation value is empty or not specified.
+                                httpMethodPaths.add("");
                             }
                             if (httpMethodPaths.isEmpty()) {
```

Comment:
IIUC, `httpMethodPaths` seems not to be empty here. Could remove this if condition?

Reply:
If this is the default, `httpMethodPaths` may be empty. This is because if the default value is in the process of getting the value, the corresponding path is not imported.

Reply:
Based on the validation and if-else condition:
- if `usePatterns` is true, `httpMethodPaths` is always not empty because of the condition of line 499.
- if `httpMethodPaths.isEmpty()` is true, an empty string is added to it.

Therefore, in line 510, `httpMethodPaths` has at least one item, in my opinion.

Reply:
Ah! I was mistaken for the number of lines. 😅
I agree to remove lines 510-513.